packages/@react-aria/grid/src/useGridCell.ts

```diff
@@ -93,7 +93,10 @@ export function useGridCell<T, C extends GridCollection<T>>(props: GridCellProps
     isVirtualized,
     focus,
     shouldSelectOnPressUp,
-    onAction: onCellAction ? () => onCellAction(node.key) : onAction
+    onAction: onCellAction ? () => onCellAction(node.key) : onAction,
+    // TODO: this will affect all grids, maybe make it specific to Table via useTableState?
+    // something like isInteractionDisabled: state.collection.size === 0
+    isDisabled: state.collection.size === 0
```

Comment:
Open to opinions here. I could make the changes in this PR to be grid wide instead of only for TableView which would make the above change valid. If we wanna only affect TableView, I can add something specific to the table state instead OR add isDisabled to useGridCell/useGridRow's props (feels iffy). A third option is that we are ok with the above change even though we are scoping this behavior to TableView, especially since TableView is the only grid that has grid cells and rows when the collection is empty